### PR TITLE
vim-patch:8.1.1336: some eval functionality is not covered by tests

### PR DESCRIPTION
**Pull request by VVKot**
Problem:    Some eval functionality is not covered by tests.
Solution:   Add a few more test cases. (Masato Nishihata, closes vim/vim#4374)
https://github.com/vim/vim/commit/17aca707f92235b6f962e637e8073162d18e6de2

### CODE REVIEW

File: test/functional/legacy/delete_spec.lua [hunk 1]
```diff
@@ -114,4 +115,16 @@ describe('Test for delete()', function()
     eq(0, eval("delete('Xdir4/Xfile')"))
     eq(0, eval("delete('Xdir4', 'd')"))
   end)
+    it('delete errors', function()
+        local err = exc_exec("call delete('''')")
+        -- Expected
+        -- neq(err:find('E474:'), nil)
+        -- Actual
+        eq(0, err)
```
Comment (VVKot) [inline test/functional/legacy/delete_spec.lua:123]: @janlazo do you know offhand what the problem here might be? Maybe `exc_exec` is not a correct function to use?

File: test/functional/legacy/delete_spec.lua [hunk 2]
```diff
@@ -114,4 +115,16 @@ describe('Test for delete()', function()
     eq(0, eval("delete('Xdir4/Xfile')"))
     eq(0, eval("delete('Xdir4', 'd')"))
   end)
+  it('delete errors', function()
+    local err = exc_exec("call delete('''')")
+    -- Expected
+    -- neq(err:find('E474:'), nil)
+    -- Actual
+    eq(0, err)
```
Comment (VVKot) [inline test/functional/legacy/delete_spec.lua:123]: @janlazo do you know what this might be? Maybe `exc_exec` is not the right helper to call?

**Review by janlazo — CHANGES_REQUESTED** [review]
Only delete_spec.lua has issues. Everything else is good.

Try creating another helper function. Maybe `:execute` suppresses the error.

File: test/functional/legacy/delete_spec.lua (hunk 2 above)
Comment (janlazo) [reply test/functional/legacy/delete_spec.lua:123]: Use pcall and `command`.

**Comment by justinmk** [comment]
closing in favor of #15502

**Pull request closed without merging**